catalog/src/main/java/org/killbill/billing/catalog/DefaultVersionedCatalog.java

```diff
@@ -190,6 +190,31 @@ private void validatePlanShape(final Plan plan, final Plan targetPlan, final Val
                                                              cur.getName(), plan.getName(), plan.getCatalog().getEffectiveDate(), targetPlan.getCatalog().getEffectiveDate()),
                                                DefaultVersionedCatalog.class, ""));
             }
+	    //Fix for https://github.com/killbill/killbill/issues/1465
+            validatePlanDuration(plan, cur, errors);
+        }
+    }
+    
+    private void validatePlanDuration(final Plan plan, final PlanPhase cur, final ValidationErrors errors) {
+        if (cur.getPhaseType().name().equals(PhaseType.EVERGREEN.name()) && !cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) {
+            errors.add(new ValidationError(String.format("EVERGREEN Phase '%s' for plan '%s' in version '%s' must have duration as UNLIMITED'",
+                    cur.getName(), plan.getName(), plan.getCatalog().getEffectiveDate()),
+                    DefaultVersionedCatalog.class, "")); 
+        }
+        if (cur.getPhaseType().name().equals(PhaseType.FIXEDTERM.name()) && cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) {
```

Comment:
This is 3 times the same check, you could greatly simplify this with:

```
if (cur.getPhaseType().name().equals(PhaseType.EVERGREEN.name()) && !cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) { /* EVERGREEN cannot have UNLIMITED duration */
...
} else if (!cur.getPhaseType().name().equals(PhaseType.EVERGREEN.name())) && cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) {  /* FIXEDTERM, TRIAL, DISCOUNT need to have UNLIMITED duration */
}
```

Reply:
Fixed.